### PR TITLE
New CI Tutorial: Build and Push to AWS ECR

### PR DESCRIPTION
**Pull request by pavanbelagatti**
Signed-off-by: Pavan Belagatti <pavan.belagatti@harness.io>

## Summary

This new CI tutorial helps the user to build and push container images to AWS ECR.

## What Type of PR is This?

- [x] Tutorial

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## HouseKeeping
Some items to keep track of. Screenshots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally

### CODE REVIEW

**Comment by pavanbelagatti** [comment]
@ravilach - Troy told me to add it. Please take a look

**Comment by ravilach** [comment]
@pavanbelagatti can you change the sign up UTM to https://app.harness.io/auth/#/signup/?module=ci&?utm_source=website&utm_medium=harness-developer-hub&utm_campaign=ci-plg&utm_content=get-started and create a tile per our Slack?

**Comment by pavanbelagatti** [comment]
@KimberlyFields can you please take a look

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+The pipeline is ready till testing and building the application. What is left is to push our built image to Amazon ECR. So, let us connect our AWS account with Harness to make sure they communicate with each other. 
+
+In the project setup, go to the ‘Connectors’ tab to connect our AWS account.
+
+![explore connectors](./static/ci-tutorial-build-push-ecr/explore_connectors.png)
+
+![connectors tab](./static/ci-tutorial-build-push-ecr/connectors.png)
+
+Select AWS from the list and add the required details.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:61]:
```suggestion
9. Select AWS from the list and add the required details. 
```

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+In the project setup, go to the ‘Connectors’ tab to connect our AWS account.
+
+![explore connectors](./static/ci-tutorial-build-push-ecr/explore_connectors.png)
+
+![connectors tab](./static/ci-tutorial-build-push-ecr/connectors.png)
+
+Select AWS from the list and add the required details. 
+![select AWS](./static/ci-tutorial-build-push-ecr/aws_connector_overview.png)
+
+There are three ways to connect your AWS account. We will use ‘AWS Access Key’ method to authenticate and connect.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:64]:
```suggestion
10. There are three ways to connect your AWS account. We will use **AWS Access Key** method to authenticate and connect.
```

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+![AWS access](./static/ci-tutorial-build-push-ecr/AWS_Access.png)
+
+Then we will connect the Harness platform as our option to connect with AWS.
+![connect AWS](./static/ci-tutorial-build-push-ecr/connect_provider.png)
+
+Once you click ‘Save and Continue’, you should see a successful connection message.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:70]:
```suggestion
After you select **Save and Continue**, you should see a successful connection message.
```

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+![cloud provider](./static/ci-tutorial-build-push-ecr/cloud_provider_success.png)
+
+Now, you have successfully connected your AWS account with Harness.
+
+It is time we need to add our last step in the CI pipeline, i,e pushing the image to Amazon ECR.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:75]:
```suggestion
It is time to add our last step in the CI pipeline; pushing the image to Amazon ECR.
```

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+Go back to your pipeline and add a step under the ‘Build and Test’ stage (under execution).
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:77]:
```suggestion
12. Go back to your pipeline and add a step under the **Build and Test** stage (under execution). 
```

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+![cloud provider](./static/ci-tutorial-build-push-ecr/cloud_provider_success.png)
+
+Now, you have successfully connected your AWS account with Harness.
+
+It is time we need to add our last step in the CI pipeline, i,e pushing the image to Amazon ECR.
+
+Go back to your pipeline and add a step under the ‘Build and Test’ stage (under execution). 
+![step library](./static/ci-tutorial-build-push-ecr/step_library.png)
+
+Select ‘Build and Push to ECR’
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:80]:
```suggestion
13. Select **Build and Push to ECR**.
```

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+It is time we need to add our last step in the CI pipeline, i,e pushing the image to Amazon ECR.
+
+Go back to your pipeline and add a step under the ‘Build and Test’ stage (under execution). 
+![step library](./static/ci-tutorial-build-push-ecr/step_library.png)
+
+Select ‘Build and Push to ECR’ 
+![build and push setup](./static/ci-tutorial-build-push-ecr/build_push_step.png)
+
+Make sure to correctly add the Region, Account ID and Image Name. Add the ‘Tags’ [I have added ‘testing’ as a Tag]. Apply changes and save the pipeline settings.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:83]:
```suggestion
14. Make sure to correctly add the Region, Account ID, and Image Name. Add the Tags [I have added **testing** as a Tag]. Apply the changes and save the pipeline settings.
```

File: tutorials/build-code/ci-build-push-to-ecr.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+Finally, your pipeline should look like this.
+![ecr tutorial](./static/ci-tutorial-build-push-ecr/ECR_tutorial.png)
+
+Basically, we are testing the application with a simple ‘npm test’ command as configured in the ‘Run’ step and pushing the built image to Amazon ECR (configured in the last step).
+
+Save and run the pipeline:)
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-build-push-to-ecr.md:90]:
```suggestion
Save and run the pipeline.
```

**Review by KimberlyFields — CHANGES_REQUESTED** [review]
Remember to use the Add suggestion to batch option to enable a single commit of all suggestions.

**Comment by pavanbelagatti** [comment]
Done, suggestions successfully applied.  
cc @ravilach & @KimberlyFields

**Review by ravilach — CHANGES_REQUESTED** [review]
@pavanbelagatti can you un-commit the `yarn.lock` file you have committed please? https://stackoverflow.com/questions/12481639/remove-file-from-latest-commit

**Comment by ravilach** [comment]
One more item if you don't mind @pavanbelagatti, can you add the tile to the All-Tutorials section please? Thanks!

**Review by KimberlyFields — APPROVED** [review]

**Comment by KimberlyFields** [comment]
@ravilach my changes were incorporated so I approved the PR. Please merge this PR when your suggested changes are implemented.

**Comment by pavanbelagatti** [comment]
@ravilach I removed the yarn file and added the tile to the All-Tutorials section.

**Review by ravilach — APPROVED** [review]

**Pull request merged**